sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -415,15 +425,20 @@ def refresh(self, **kwargs) -> None:
             with self._update_lock:
                 for (key, label), etag in self._refresh_on.items():
                     updated_sentinel = self._client.get_configuration_setting(
-                        key=key, label=label, etag=etag, match_condition=MatchConditions.IfModified, **kwargs
+                        key=key,
+                        label=label,
+                        etag=etag,
+                        match_condition=MatchConditions.IfModified,
+                        headers=_get_headers("Watch", uses_key_vault=self._uses_key_vault, **kwargs),
+                        **kwargs
                     )
                     if updated_sentinel is not None:
                         logging.debug(
                             "Refresh all triggered by key: %s label %s.",
                             key,
                             label,
                         )
-                        self._load_all(**kwargs)
+                        self._load_all(headers=_get_headers("Watch", uses_key_vault=self._uses_key_vault, **kwargs))
```

Comment:
Are the request types standardized?
I understand that the request to get updated sentinel keys is a watch request. But should the request to load all during refresh have a request type of e.g., "Refresh"?

Reply:
Yes they are.